src/lib.rs

```diff
@@ -159,14 +159,19 @@ pub unsafe extern "C" fn WrapperDecompressImageEx(
 ) -> i32 {
```

Comment:
I think that use_16bit_dc_estimate param to the function should become use_scalar_lepton and to control both flags - use_16bit_dc_estimate and use_16bit_adv_predict. Also, is there are a reason to keep these flags separated in EnabledFeatures?

Reply:
unfortunately for files that were compressed by the older rust libraries we need the flags seperate because they were encoded with use_16bit_dc_estimate = false and use_16bit_adv_predict = true. For files that are encoded by the new version, we record which parameters were used so the decoder will always be right.

Reply:
Got it. I missed that this change is not needed for back-compat, but it improves the compression for future compressed files and contains a small format change to support it.